engine/function/src/templates/Numeric.ftl

```diff
@@ -0,0 +1,2446 @@
+    }
+
+    /**
+     * Returns the closest integer to the argument.  If the argument is NaN, the result is 0.  If the argument is greater
+     * than {@code Integer.MIN_VALUE}, {@code Integer.MIN_VALUE} is returned.  If the argument is less than {@code Integer.MAX_VALUE},
+     * {@code Integer.MAX_VALUE} is returned.
+     *
+     * @param value value.
+     */
+    public static long round(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_LONG;
+        }
+
+        return Math.round(value);
+    }
+
+    /**
+     * Returns the signum function.
+     *
+     * @param value value.
+     * @return signum function.
+     */
+    public static int signum(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_INT;
+        }
+
+        return Integer.signum((int)value);
+    }
+
+    /**
+     * Returns the sine.
+     *
+     * @param value value.
+     * @return sine.
+     */
+    public static double sin(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.sin(value);
+    }
+
+    /**
+     * Returns the square root.
+     *
+     * @param value value.
+     * @return square root.
+     */
+    public static double sqrt(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.sqrt(value);
+    }
+
+    /**
+     * Returns the tangent.
+     *
+     * @param value value.
+     * @return tangent.
+     */
+    public static double tan(${pt.primitive} value) {
+        if (isNull(value)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.tan(value);
+    }
+
+
+    <#if pt.valueType.isFloat >
+
+    /**
+     * Returns the lower bound of the bin containing the value.
+     *
+     * @param value value.
+     * @param interval bin width.
+     * @return lower bound of the bin containing the value.
+     */
+    public static ${pt.primitive} lowerBin(${pt.primitive} value, ${pt.primitive} interval) {
+        if (value == ${pt.null} || interval == ${pt.null}) {
+            return ${pt.null};
+        }
+
+        return (${pt.primitive}) (interval * Math.floor(value / interval));
+    }
+
+    <#else>
+
+    /**
+     * Returns the lower bound of the bin containing the value.
+     *
+     * @param value value.
+     * @param interval bin width.
+     * @return lower bound of the bin containing the value.
+     */
+   public static ${pt.primitive} lowerBin(${pt.primitive} value, ${pt.primitive} interval) {
+        if (value == ${pt.null} || interval == ${pt.null}) {
+            return ${pt.null};
+        }
+
+        if ( interval <= 0 ) {
+            throw new IllegalArgumentException("Interval is not positive: " + interval);
+        }
+
+        final long d = ((long)value) / ((long)interval);
+        final long m = ((long)value) % ((long)interval);
+        final long r = (m != 0 && value < 0) ? d - 1 : d;
```

Comment:
`value < 0` implies `m < 0` so we could delete `m != 0` from this if statement.
(I think)

Reply:
Actually, `value < 0` does not imply `m < 0`.  See here:

<img width="1307" alt="image" src="https://user-images.githubusercontent.com/5250374/165998718-a28744ed-06af-497f-9fd4-139d7bf3f5a3.png">

Removing `m != 0` results in unit test failures.

Reply:
um, oops. thanks for verifying. sorry.